src/inmanta/parser/plyInmantaLex.py

```diff
@@ -56,13 +55,56 @@
     "else",
     "elif",
 ]
-literals = [":", "[", "]", "(", ")", "=", ",", ".", "{", "}", "?", "*"]
+literals = [":", "[", "]", "(", ")", "=", ",", ".", "{", "}", "?", "*", "%"]
 reserved = {k: k.upper() for k in keyworldlist}
 
 # List of token names.   This is always required
-tokens = ["INT", "FLOAT", "ID", "CID", "SEP", "STRING", "MLS", "CMP_OP", "REGEX", "REL", "PEQ", "RSTRING", "FSTRING"] + sorted(
-    list(reserved.values())
-)
+tokens = [
+    "INT",
+    "FLOAT",
+    "ID",
+    "CID",
+    "SEP",
+    "STRING",
+    "MLS",
+    "CMP_OP",
+    "REGEX",
+    "MATCHING",
+    "REL",
+    "PEQ",
+    "RSTRING",
+    "FSTRING",
+    "PLUS_OP",
+    "MINUS_OP",
+    "DIVISION_OP",
+    "DOUBLE_STAR",
+] + sorted(list(reserved.values()))
+
+
+def t_REGEX(t: lex.LexToken) -> lex.LexToken:  # noqa: N802
+    r"matching[\s]+/([^/\\\n]|\\.)+/"
+    index_first_slash_char = t.value.index("/")
+    regex_with_slashes_as_str = t.value[index_first_slash_char:]
+    regex_as_str = regex_with_slashes_as_str[1:-1]
+    value = Reference("self")  # anonymous value
+    try:
+        expr = Regex(value, regex_as_str)
+        t.value = expr
+        return t
+    except RegexError as error:
+        t.value = regex_with_slashes_as_str
+        end = t.lexer.lexpos - t.lexer.linestart + 1
+        (s, e) = t.lexer.lexmatch.span()
+        start = end - (e - s) + index_first_slash_char
+
+        r: Range = Range(t.lexer.inmfile, t.lexer.lineno, start, t.lexer.lineno, end)
+        raise ParserException(r, t.value, f"Regex error in {t.value}: '{error}'")
+
+
+def t_MATCHING(t: lex.LexToken) -> lex.LexToken:  # noqa: N802
```

Comment:
Same note about the whitespace. Though I have to admit it's more difficult here. Something like the following is allowed
```
typedef x as string matching(
    self == "hello"
)
```

But of course something like `typedef x as string matchingself == "hello"` isn't because then `matchintself` is a single token.

I'm a bit too far removed from the lexer at the moment to know if this is easy to address, or whether we should consider just accepting the breaking change on the edge case.

Reply:
not too concerned. Might make sense to ask @edvgui if we generate something like this

Reply:
Everything that we generate using `inmanta-module-factory` (everything that I am aware off) uses this template for typedef expressions, so there is always a space after `matching`: https://code.inmanta.com/solutions/tools/inmanta-module-factory/-/blob/2ebb08e917bab274cb7276a76305f2929f37960b/src/inmanta_module_factory/inmanta/typedef.py#L74
```
stmt = f"typedef {self.name} as {base_type_reference} matching {self.constraint}"
```

Reply:
I think the lexer is set to ignore whitespace between tokens, so it is a moot point?

Reply:
I don't think so because we now make `matching /regex/` a single token, which is the change I'm talking about: the lexer used to ignore whitespace, we require one.